### ISSUE TITLE
index 544 is out of bounds for axis 1 with size 50

### ISSUE THREAD

**Issue by ricardobnjunior** [priya-dwivedi/Deep-Learning#35]
Dear Dr Priya,
I am having an error when I try execute the train step. The error is: index 544 is out of bounds for axis 1 with size 50.

I tried on Windows and Unbutu and the error it occurred in both.

Can you help me please?

![untitled-1](https://user-images.githubusercontent.com/18532570/45068284-1dbbc880-b09d-11e8-9b4a-8de5b659df7a.jpg)

**Comment by emanuelevivoli** [comment]
hi @ricardobnjunior , I was having the same error.
This appens because you try to label with categorical a number of class of 544 to 50 writers.
You have to take images from only 50 classes.
I write this in order to take first 50 classes writers:
```python
app_1, app_2 = [ ], [ ]
for x, y in zip(img_files, img_targets):
    if int(y) < 50:
        app_1.append(x)
        app_2.append(y)
img_files, img_targets = np.asarray(app_1), np.asarray(app_2)
```

use this with:
```python 
encoder = LabelEncoder()
encoder.fit(img_targets)
# encoded_Y = [0, 1, 2, 3, 4, ... , 49]
encoded_Y = encoder.transform(img_targets)
```
before `train_test_split()` function.

Let me know if it will work for you.
by, Immanuel.
